### PR TITLE
Dj/check

### PR DESCRIPTION
**Pull request by satra**
i think your errors were related to updates i made to master. test_10a passes fine one my computer.

### CODE REVIEW

File: pydra/engine/tests/test_satraexample.py [hunk 1]
```diff
@@ -360,25 +360,24 @@ def test_11(plugin):
...
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert results.output.out == [3, 4]
```
Comment (satra) [inline pydra/engine/tests/test_satraexample.py:380]: technically there should also be in assert on this that `wf.state` has some final state.

**Comment by djarecka** [comment]
@satra - so which part of this pr is the fix? I have to say that I spent some time on it, but I didn't understand the issue...

**Comment by satra** [comment]
the saving of error files so the fix is really in getstate and setstate of the result object

**Comment by djarecka** [comment]
hmm... and this is why I couldn't do `deepcopy(self)` in `to_task`?

**Comment by satra** [comment]
`result.output` was `None` for some error state and the data were not being saved appropriately. so loadresult was returning a weird error (i also removed the try except on EOF). this in turn made deepcopy fail. 

you can do deepcopy now, right?

**Comment by djarecka** [comment]
yes, now deepcopy works

**Pull request merged**